forgerock-openbanking-upgrade/src/main/java/com/forgerock/openbanking/upgrade/UpgradeApplication.java

```diff
@@ -81,12 +90,26 @@ public void run(String... args) {
         }
     }
 
-    public abstract EntitiesVersion getPreviousVersion();
+    public abstract EntitiesVersion getStoredVersion();
 
     public abstract EntitiesVersion saveEntitiesVersion(EntitiesVersion entitiesVersion);
 
+    public abstract void deleteEntitiesVersion(EntitiesVersion entitiesVersion);
+
     private String getVersion() {
         return buildProperties.getVersion();
     }
 
+    protected List<EntitiesVersion> cleanDuplicatedVersions(List<EntitiesVersion> allVersions) {
+        if (!allVersions.isEmpty()) {
+            List<EntitiesVersion> uniques = allVersions.stream()
+                    .collect(collectingAndThen(toCollection(() -> new TreeSet<>(comparing(EntitiesVersion::getVersion))),
+                            ArrayList::new));
+            allVersions.stream().filter(version -> !uniques.contains(version)).collect(Collectors.toList())
+                    .forEach(entitiesVersion -> deleteEntitiesVersion(entitiesVersion));
```

Comment:
If we are going to delete duplicate versions are we certain that the version that will be deleted is the correct one? Which one should be deleted? the latest or earliest version or does the date not matter? is it possible to have 2 identical versions with different upgrade statuses? if so which one should be deleted?